.github/workflows/build-binaries.yml

```diff
@@ -29,46 +33,67 @@ jobs:
         with:
           fetch-depth: 0
 
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+          cache: true
+
+      - name: Set up Rust
+        uses: dtolnay/rust-toolchain@stable
+      
+      - name: Cache Rust dependencies
+        uses: Swatinem/rust-cache@v2
+        with:
+          workspaces: |
+            vm/rust
+            core/rust
+            starknet/compiler/rust
+
       - name: Get latest tag
         run: echo "TAG=$(git describe --tags)" >> $GITHUB_ENV
 
       - name: Get artifact name
-        run: echo "ARTIFACT_NAME=juno-${{ env.TAG }}-${{ runner.os }}-$(uname -m)" >> $GITHUB_ENV
+        run: |
+          OS_NAME=$([ "${{ runner.os }}" == "macOS" ] && echo "darwin" || echo "linux")
+          echo "ARTIFACT_NAME=juno-${{ env.TAG }}-${OS_NAME}-${{ matrix.arch }}" >> $GITHUB_ENV
```

Comment:
why relying on `matrix.arch` which is hard-coded instead of `uname -m` which is dynamic and will always reflect the correct arch?